camera_system/camera_calibrator.py

```diff
@@ -50,45 +74,90 @@ def make_game_area_info(self, is_left_course: bool, game_save_path="game_course.
         """
         # ゲームエリア画像を取得
         game_area_img = self.__camera_interface.capture_frame(game_save_path)
+
         # 6色変換
         color_save_path = "color_" + game_save_path
         self.__color_changer.change_color(game_area_img, color_save_path)
 
-        # カラーIDを格納する配列を宣言
-        block_color_list = []
-        base_color_list = []
-        bonus_color = []
+        # 色IDを格納する配列を宣言
+        block_color_list = np.zeros(CameraCalibrator.__COLOR_BLOCK_NUM)
+        base_color_list = np.zeros(CameraCalibrator.__BASE_BLOCK_NUM)
+        bonus_color = np.zeros(CameraCalibrator.__BONUS_BLOCK_NUM)
 
-        # ブロック置き場
+        # ブロックの色を調べる領域のピクセル数を求める
+        area_pixel_sum = CameraCalibrator.__SEARCH_AREA_XSIZE*CameraCalibrator.__SEARCH_AREA_YSIZE
+
+        # ブロック置き場上のカラーブロックの色IDを求める
         for i, point in enumerate(self.__coord.block_point):
-            # 最頻値を求めてブロックの色を判定
-            color_id = self.__color_changer.calculate_mode_color(point[0],
-                                                                 point[1],
-                                                                 CameraCalibrator.MODE_AREA_XSIZE,
-                                                                 CameraCalibrator.MODE_AREA_YSIZE)
-            block_color_list.append(Color(color_id))
-            print("ブロック置き場%d:%s" % (i, Color(color_id).name))
-        # ベースサークル置き場
+            # ブロック上の領域に存在する色の種類とピクセル数を取得
+            color_uniqs, color_pixel_sum = self.__color_changer.search_color(
+                point[0],
+                point[1],
+                CameraCalibrator.__SEARCH_AREA_XSIZE,
+                CameraCalibrator.__SEARCH_AREA_YSIZE)
+            # 色の種類とピクセル数を配列に格納
+            # 第2引数はindexと色IDを合わせるために-1
+            # 第3引数は色を求める際に領域に対する割合で比較できるように各色のピクセル数÷全体のピクセル数
+            np.put(self.count_point[i], color_uniqs-1,
+                   color_pixel_sum/area_pixel_sum)
+        # 認識したブロックの数を把握するための配列
+        color_count = np.zeros(CameraCalibrator.__VALIDITY_COLOR_NUM)  # (赤、黄、緑、青)
+        # 各ブロックの領域に対する色の割合が高い順に色IDを割り振る
+        for i in range(block_color_list.shape[0]):
```

Comment:
```suggestion
        for _ in range(block_color_list.shape[0]):
```

使わない変数は_を使う慣習があるらしい。
使わないことを明示できるので、こだわりがなければ変えてほしいです